src/traces/sankey/render.js

```diff
@@ -172,7 +172,7 @@ function sankeyModel(layout, d, traceIndex) {
         valueFormat: valueFormat,
         valueSuffix: valueSuffix,
         textFont: textFont,
-        translateX: domain.x[0] * width + layout.margin.l,
+        translateX: domain.x[0] * layout.width + layout.margin.l,
```

Comment:
Man, sankey is so confusing... I was *convinced* this must be wrong but no, you got it exactly right, sankey just remapped `_size` into something that looks like `_fullLayout` and even called it `layout` for further obfuscation...
https://github.com/plotly/plotly.js/blob/ff3c324bc09029766b9e4b7ab517b83166a7e363/src/traces/sankey/plot.js#L257-L266

If you ever feel like untangling a little of this, please do!